nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/DependencyConflictResolver.cs

```diff
@@ -467,14 +467,6 @@ public async Task<string> UpdateVersion(List<PackageToUpdate> existingPackages,
                             // Loop from the current version to the latest version, use next patch as a limit (unless there's a limit) so it doesn't look for versions that don't exist
                             for (NuGetVersion version = currentVersionParent; version <= latestVersion; version = NextPatch(version, versions))
                             {
-                                NuGetVersion nextPatch = NextPatch(version, versions);
-
-                                // If the next patch is the same as the currentVersioon, then the update is a Success
-                                if (nextPatch == version)
-                                {
-                                    return "Success";
```

Comment:
This was the early exit bug and is functionally equivalent to the `return` 17 lines down, but after we've assigned a new version.